src/language/plain-text.md

```diff
@@ -37,3 +53,78 @@ script.
   else
     console.log('use pug')
 ```
+
+You can also create a dot block of plain text *after* other tags within the parent tag.
+
+```pug-preview
+div
+  p This text belongs to the paragraph tag.
+  br
+  .
+    This text belongs to the div tag.
+```
+
+## Whitespace Control
+
+Managing the whitespace of the rendered HTML is the trickiest part about learning Pug. By default, Pug removes spaces from the beginning and end of the lines for both inline-level tags and plain text lines and blocks. The value here is that it gives you full control over whether inline tags and/or plain text should touch. It even lets you place tags in the middle of words.
+
+```pug-preview
+| You put the em
+em pha
+| sis on the wrong syl
+em la
+| ble.
+```
+
+The trade-off is that it *requires* you to think about and take control over whether tags and text touch. Don't worry, though, you'll get the hang of it soon enough.
+
+If you need the text and/or inline-level tags to touch --- perhaps you need a period to appear outside the hyperlink at the end of a sentence --- this is easy, as it's basically what happens unless you tell Pug otherwise.
+
+```pug-preview
+a ...sentence ending with a link
+| .
+```
+
+If you need to *add* space, you have a few options:
+
+Depending on where you need the whitespace, you could add an extra space at the beginning of the plain text line (after the pipe for piped text). Or you could add a trailing space at the *end* of the plain text line. Less often, you might need extra space *inside the tag*, but it's the same idea --- extra spaces at the beginning of the text (after the tag), or at the end of the line. This may require configuring your code editor *not* to strip trailing spaces when you save, and you will need to make sure all contributors to your project do the same.
+
+**NOTE** the trailing and leading spaces here:
+
+```pug-preview
+| Hey, check out 
+a(href="http://example.biz/kitteh.png") this picture
+|  of my cat!
+```
+
+You could add one or more "empty" piped line --- a pipe with either spaces or nothing after it. This will insert whitespace in the rendered HTML.
+
+```pug-preview
+| Don't
+|
+em touch
+|
+| me!
+```
+
+Another trick is to use string [interpolation](interpolation.html): `=` can be used to add a literal space or newline character to the rendered HTML.
```

Comment:
I think we should probably just remove it.

For the attributes example, what I'd like to do is replace the pug pretty mode with an html formatter based on http://demos.forbeslindesay.co.uk/htmlparser2/ so that we can:

1. deprecate pretty mode, since it's the cause of a lot of people struggling to work out why pug sometimes does one thing and sometimes does something else.
2. change the behaviour of pretty printing to suit our documentation, without worrying about breaking anybody's website.